framework/render/tasks.py

```diff
@@ -61,20 +62,26 @@ def _build_rendered_html(download_url, cache_path, temp_path):
     ensure_path(os.path.split(temp_path)[0])
     ensure_path(os.path.split(cache_path)[0])
 
-    save_to_file_or_error(download_url, temp_path)
+    with codecs.open(cache_path, 'w', 'utf-8') as render_result_cache:
+        try:
```

Comment:
I decided to keep them loose so that the logic is a bit more clear.

Opinions there?